include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -151,6 +177,28 @@ def VariableOp : MooreOp<"variable", [
   }];
 }
 
+def NetOp : MooreOp<"net", [
+  MemoryEffects<[MemAlloc]>,
+  DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+  OptionalTypesMatchWith<"assigned value and variable types match",
+    "result", "assignment", "$_self">,
+]> {
+  let summary = "A net declaration";
+  let description = [{
+    See IEEE 1800-2017 § 6.7 "Net declarations".
+  }];
+  let arguments = (ins
+    StrAttr:$name,
+    StrAttr:$kind,
+    Optional<UnpackedType>:$assignment
+  );
+  let results = (outs UnpackedType:$result);
+  let assemblyFormat = [{
+    `` custom<ImplicitSSAName>($name) $kind ($assignment^)? attr-dict
```

Comment:
nit: Do we need `? 
```suggestion
    custom<ImplicitSSAName>($name) $kind ($assignment^)? attr-dict
```

Reply:
I did not notice, Thank you to point it out !